### PR TITLE
Prefer the displayName rather than the fullName for prettier display

### PR DESCRIPTION
**Pull request by alecharp**
# Description

See [JENKINS-38042](https://issues.jenkins-ci.org/browse/JENKINS-38042).

There is no unit test as the displayed value of the `Link` in `PipelineRowItem` is not tested at this point. 

I "manually" validated the change: I created a job which I named `my-ugly-project` and set its __Display name__ to `My Beautiful project`. I also reacted another project without changing the __Display name__ to validate it didn't affect the current behavior. But as the `displayName` is filled with the `fullName` when it's not specified, then the front part of this plugin has no impact on this.

I couldn't run a complete build of the project as the current `master` branch has code inspections issues (lines being too longs) even before I coded my patch.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
thanks @alecharp - there are test(s) failing - which do cover this code. If you adjust them to pass then this might be good to go!

**Comment by alecharp** [comment]
oh crap. I only looked at `src/test/js/pipelines-spec.js` which doesn't test the content of the link. I'm working on it.

**Comment by scherler** [comment]
@alecharp there had been https://issues.jenkins-ci.org/browse/JENKINS-35901 where the fullName had been preferred of displayName. @i386 or @cliffmeyers would know the specifics but I guess it has something to do with multibranch and/or folder projects. Did you try with multibranch projects?

**Comment by alecharp** [comment]
@scherler the "normal" pipeline job doesn't have any "Display" field in it's configuration. Only the MultiBranch Pipeline has. By the way, the `fullName` and the `displayName` have the same value if no specific `displayName` is configured. 

If the job creator did specified a `displayName` we should use it. It meant to have a "nicer" human readable value. If the blueocean plugin is about to have a better / nicer UX, this should be part of the plugin, don't you think? Maybe the [JENKINS-35901](https://issues.jenkins-ci.org/browse/JENKINS-35901) was created when only `name` was provided in the REST impl.

**Comment by scherler** [comment]
@alecharp I took screenshots of what we have now

screenshot from 2016-10-10 15-29-08

you can see the "folder" structure and the project name.

screenshot from 2016-10-10 15-30-04

Applying your patch you will loose the path and AFIAK @i386 wants that path. However maybe you can rewrite the patch to include the path AND the display name.

**Comment by alecharp** [comment]
yes I confirm this. Working on it.

**Comment by michaelneale** [comment]
ok thanks @alecharp and thanks @scherler for clarifying this

File: blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx [hunk 1]
```diff
@@ -44,7 +44,8 @@ export default class PipelineRowItem extends Component {
         const pullRequestsURL = `${baseUrl}/pr`;
         const activitiesURL = `${baseUrl}/activity`;
 
-        const formattedName = displayName ? displayName.split('/').join(' / ') : '';
+        const pathInJob = fullName.split('/').slice(0, -1).join(' / ');
+        const formattedName = `${pathInJob? `${pathInJob} / ` : ''}${displayName ? displayName.split('/').join(' / ') : ''}`;
```
Comment (scherler) [inline blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx]: ``` ${displayName ? displayName.split('/').join(' / ') : ''}``` should be ``` ${displayName}```

**Comment by scherler** [comment]
:bee: when CI passes

**Pull request merged**